### ISSUE TITLE
Window overlay not showing properly with a specific monitor setup

### ISSUE THREAD

**Issue by uvera** [instantOS/instantWM#68]
Basically I've run into the issue with the overlay showing improperly with this monitor setup
(Vertical: 24in FHD, Horizontal 32in QHD)

![20210108150355](https://user-images.githubusercontent.com/5598427/104024039-cecd5300-51c2-11eb-91a8-a24fe81030cc.png)

The issue is the overlay on the vertically higher monitor is showing properly like this:

![20210108150300](https://user-images.githubusercontent.com/5598427/104024097-e278b980-51c2-11eb-9e7c-aa63ac69f39f.png)

But with the vertically smaller monitor it's shown like this:

![20210108150307](https://user-images.githubusercontent.com/5598427/104024137-ee647b80-51c2-11eb-83d3-c8ae1ce7d834.png)

Temporary workaround I did is:
```
diff --git a/instantwm.c b/instantwm.c
index 6985fdf..36e563e 100644
--- a/instantwm.c
+++ b/instantwm.c
@@ -460,7 +460,19 @@ showoverlay() {
 		XRaiseWindow(dpy, c->win);
         switch (selmon->overlaymode) {
             case 0:
-			    animateclient(c, c->x, ( selmon->showbar ? bh : 0 ), 0, 0, 15, 0);
+                {
+                    Monitor *iter_mon = selmon;
+                    int calc_val = 0;
+                    while (iter_mon && iter_mon->next)
+                    {
+                        int iter_mon_wh = iter_mon->wh;
+                        int iter_mon_next_wh = iter_mon->next->wh;
+                        int calculation = iter_mon_next_wh > iter_mon_wh ? (iter_mon_next_wh - iter_mon_wh) / 2 : 0;
+                        calc_val = calculation > calc_val ? calculation : calc_val;
+                        iter_mon = selmon->next;
+                    }
+                    animateclient(c, c->x, ( selmon->showbar ? bh + calc_val : calc_val ), 0, 0, 15, 0);
+                }
                 break;
             case 1:
                 animateclient(c, selmon->mx + selmon->mw - c->w, 40, 0, 0, 15, 0);
```

The issue with this patch is the Monitor struct being a single linked list, so when the primary monitor is the vertical one, the horizontal one doesn't have the next node and can't calculate the height offset for overlay. I'm currently not experienced enough to find a workaround.

**Comment by con-f-use** [comment]
Possibly related to #60

**Comment by paperbenni** [comment]
Jesus christ that's a complicated way to go about this :laughing:  
The issue was just that the position of the monitor didn't get added to the target position when showing the overlay
![grafik](https://user-images.githubusercontent.com/15818888/104041298-1eb71480-51d9-11eb-83c7-2935352cc50b.png)
I issued a fix, pretty sure the issue should be gone now.  
Unfortunately I don't think this will fix #60

**Comment by uvera** [comment]
Tested the changes from the master branch, seems the issue is gone, thanks!
